api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java

```diff
@@ -125,49 +125,21 @@ public BatchRecorder newMeasureBatchRecorder() {
   }
 
   @Override
-  public LabelSet createLabelSet(String k1, String v1) {
-    Utils.checkNotNull(k1, "k1");
-    Utils.checkNotNull(v1, "v1");
-    return NoopLabelSet.INSTANCE;
-  }
-
-  @Override
-  public LabelSet createLabelSet(String k1, String v1, String k2, String v2) {
-    Utils.checkNotNull(k1, "k1");
-    Utils.checkNotNull(v1, "v1");
-    Utils.checkNotNull(k2, "k2");
-    Utils.checkNotNull(v2, "v2");
-    return NoopLabelSet.INSTANCE;
-  }
-
-  @Override
-  public LabelSet createLabelSet(String k1, String v1, String k2, String v2, String k3, String v3) {
-    Utils.checkNotNull(k1, "k1");
-    Utils.checkNotNull(v1, "v1");
-    Utils.checkNotNull(k2, "k2");
-    Utils.checkNotNull(v2, "v2");
-    Utils.checkNotNull(k3, "k3");
-    Utils.checkNotNull(v3, "v3");
-    return NoopLabelSet.INSTANCE;
-  }
-
-  @Override
-  public LabelSet createLabelSet(
-      String k1, String v1, String k2, String v2, String k3, String v3, String k4, String v4) {
-    Utils.checkNotNull(k1, "k1");
-    Utils.checkNotNull(v1, "v1");
-    Utils.checkNotNull(k2, "k2");
-    Utils.checkNotNull(v2, "v2");
-    Utils.checkNotNull(k3, "k3");
-    Utils.checkNotNull(v3, "v3");
-    Utils.checkNotNull(k4, "k4");
-    Utils.checkNotNull(v4, "v4");
+  public LabelSet createLabelSet(String... keyValuePairs) {
+    Utils.checkArgument(
+        keyValuePairs.length % 2 == 0,
+        "You must provide an even number of key/value pair arguments.");
+    for (int i = 0; i < keyValuePairs.length; i += 2) {
+      String key = keyValuePairs[i];
+      Utils.checkNotNull(key, "You cannot provide null keys for LabelSet creation.");
+    }
     return NoopLabelSet.INSTANCE;
   }
 
   @Override
   public LabelSet createLabelSet(Map<String, String> labels) {
```

Comment:
Do we need this method anymore?

Reply:
We could definitely get rid of it, if we have varargs, but I suspect there might be API usages that would want to have a static Map around to use. I'd like to get some feedback on the varargs option before we remove the Map option. We could also get rid of the empty method, since you can just call the varargs one with no arguments to get the empty LabelSet back.

Reply:
Definitely the empty api does not make that much sense. I am for removing the empty and maybe ask some users for feedback for the map?

Reply:
Sounds good to me. I'll make those changes and update the PR.

Reply:
Do you want to do the changes in this PR or separately?

Reply:
updated in this PR already. :)